tools/release-ci/pkg/pullrefs/cloneref.go

```diff
@@ -0,0 +1,92 @@
+package pullrefs
+
+import (
+	"fmt"
+	"io"
+	"os/exec"
+)
+
+// Runner is an object that knows how to keep track of a list
+// of commands and execute them
+type Runner interface {
+	Add(cmd string, args ...string)
+	AddAt(dir, cmd string, args ...string)
+	Run() error
+}
+
+// CloneRef clones a SourceRef in the specified directory
+func CloneRef(ref SourceRef, dir string, existing bool, runner Runner) error {
+	if len(ref.RepositoryURL) == 0 {
+		return fmt.Errorf("a repository URL must be specified")
+	}
+
+	if !existing {
+		runner.Add("mkdir", "-p", dir)
```

Comment:
sure... can make it a func ... all I wanted was something to let me make the code clearer so I don't have a lot of error checking cluttering things.